### PR TITLE
[MRG] convert all bids_basename to bids_path.

### PR DESCRIPTION
**Pull request by adam2392**
PR Description
--------------

Entirely cosmetic and removed this one unused function.

Converts all `bids_basename` instances to `bids_path` wherever it makes sense.

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"
- [ ] Commit history does not contain any merge commits

### CODE REVIEW

File: examples/convert_eeg_to_bids.py [hunk 1]
```diff
@@ -175,8 +175,8 @@
 event_id = {'rest': 0}
 
 # Now convert our data to be in a new BIDS dataset.
-bids_basename = BIDSPath(subject=subject_id, task=task)
-write_raw_bids(raw, bids_basename, bids_root, event_id=event_id,
+bids_path = BIDSPath(subject=subject_id, task=task)
+write_raw_bids(raw, bids_path, bids_root, event_id=event_id,
```
Comment (jasmainak) [inline examples/convert_eeg_to_bids.py:179]: we still need to pass `bids_root`? maybe time to deprecate that parameter?

Comment (adam2392) [reply examples/convert_eeg_to_bids.py:179]: I agree. Shall I do that in a next round? I think this one being purely cosmetic has a large diff, so would prefer just getting it out of the way, so the refactoring of kwarg arguments is small diff and easy to review.

File: mne_bids/path.py [hunk 2]
```diff
@@ -998,46 +998,6 @@ def _find_best_candidates(params, candidate_list):
     return best_candidates
 
 
-def _get_bids_fname_from_filesystem(*, bids_basename, bids_root, sub, ses,
-                                    modality):
-    if modality is None:
-        modality = _infer_modality(bids_root=bids_root, sub=sub, ses=ses)
-
-    data_dir = make_bids_folders(subject=sub, session=ses,
-                                 modality=modality, make_dir=False)
```
Comment (adam2392) [inline mne_bids/path.py:1007]: unused function leftover. idk why flake8 didn't catch :(

**Comment by hoechenberger** [comment]
I cannot post any inline comments anymore somehow, but just re `bids_root` in some function signatures, earlier today I made this 2-point todo for exactly the same thing:

- read_raw_bids() and write_raw_bids() should drop the bids_basename and bids_root kwargs in favor of bids_path
- get_matched_empty_room() should do the same

Let's do this in a separate PR after this one. Let's try to keep diffs small :)

**Review by agramfort — APPROVED** [review]
@hoechenberger merge if happy

**Review by hoechenberger — APPROVED** [review]

**Comment by hoechenberger** [comment]
Thank you, @adam2392

**Comment by adam2392** [comment]
> I cannot post any inline comments anymore somehow, but just re `bids_root` in some function signatures, earlier today I made this 2-point todo for exactly the same thing:
> 
> * read_raw_bids() and write_raw_bids() should drop the bids_basename and bids_root kwargs in favor of bids_path
> * get_matched_empty_room() should do the same
> 
> Let's do this in a separate PR after this one. Let's try to keep diffs small :)

So regarding this, is this the best thing to do? 

I'm just asking because right now we allow a user to pass in the directory for the bids_root and also a string denoting the basename. If we refactor to demand a BIDSPath, then should we essentially error out if they do not pass one in asking them to use `BIDSPath` and `get_entities_from_fname`? 

Essentially no reading/writing occurs unless they use `BIDSPath`?

**Comment by hoechenberger** [comment]
Yes we should demand a BIDSPath with a number of required entities and the modality set.

Not sure what you mean with get_entities_from_fname though

**Pull request merged**